src/peakrdl_sv/node.py

```diff
@@ -9,8 +9,14 @@
 
 
 class Node(UserList):
-    def __init__(self, node: AddrmapNode | RegfileNode | RegNode | FieldNode) -> None:
+    def __init__(
+        self,
+        node: AddrmapNode | RegfileNode | RegNode | FieldNode,
+        parent: AddrmapNode | RegfileNode | RegNode | None,
+    ) -> None:
         self.node = node
+        if parent is not None:
```

Comment:
Should we have an assert here to check that if the parent is None, then the node must be an `AddressMap`.  Might catch some broken usage.

Reply:
That makes sense.